system-test/storage.ts

```diff
@@ -906,12 +907,13 @@ describe('storage', () => {
 
     // Normalization form D: an ASCII character followed by U+0301 combining
     // character; URL should end with Caf%C3%A9
-    it('should not perform normalization form D', () => {
+    it('should not perform normalization form D', async () => {
       const name = 'Cafe\u0301';
-      const file = bucket.file(name);
-
       const expectedContents = 'Normalization Form D';
 
+      const file = bucket.file(name);
+      await file.save(expectedContents);
+
       return file
```

Comment:
Should this logic be switched to use `await` as well?

Reply:
Which logic are you referring to here? `bucket.file(` or the `return` statement?

Reply:
`file.get()` and `receivedFile.downLoad()` functions

Reply:
What is the difference between `.then()` and `await`? From what I understand, `then` will wait for the prior code to return before executing. Is this understanding incorrect?

Reply:
`await X()` suspends execution until `X`'s async call completes.
`promise.then(X)` adds the `X` to the callback chain and continues the current execution.
```ts
      function greeting() {
        return new Promise(resolve => {
          resolve(console.log('Hi'));
        });
      }
      function hru() {
        console.log('How are you?');
      }
      function cu() {
        console.log('See you later');
      }
```
```ts
     greeting().then(() => hru());
     cu();
```
will print
`Hi`
`See you later`
`How are you?`
vs
```ts
      await greeting();
      hru();
      cu();
```
will print
`Hi`
`How are you?`
`See you later`

Reply:
This is a super helpful explanation, thank you so much! I really appreciate it. In the case of our return statement here, can you please clarify your concern? In your first example above, `Hi` will always print before `How are you?`, correct? And so in the case of our return statement, it will wait for `file.get()` before the next step and `receivedFile.download()` before the last step.

Reply:
The concern is that it may continue to the the next test after firing `file.get()` before getting to `.then()`'s and not getting to the `assert`s.